File: Dashboard/Acme.NativeSQLiteDataAdapter/SQLiteDataProvider.cpp
#include "stdafx.h"
#include "SQLiteDataProvider.h"
#include "SQLiteRepository.h"

using namespace System;
using namespace System::Data;

using namespace Acme::Data;
using namespace Acme::NativeSQLiteDataAdapter;

String^ SQLiteDataProvider::Name::get()
{
    return gcnew String("Native SQLite3 Data Provider");
}

bool SQLiteDataProvider::SupportsEncryption::get()
{
    return false;
}

IDataRepository^ SQLiteDataProvider::Open(System::String^ databaseFilePath)
{
    return gcnew SQLiteRepository(databaseFilePath);
}

IDataRepository^ SQLiteDataProvider::Open(System::String^ databaseFilePath, System::String^ password)
{
    throw gcnew NotSupportedException();
}

File: Dashboard/Acme.ManagedSQLiteDataAdapter/SQLiteDataProvider.cs
using Acme.Data;
using System;

namespace Acme.ManagedSQLiteDataAdapter
{
    public class SQLiteDataProvider : DataProvider
    {
        public override string Name
        {
            get { return "Managed SQLite 3 Data Provider"; }
        }

        public override bool SupportsEncryption
        {
            get { return true; }
        }

        public override IDataRepository Open(string databaseFilePath)
        {
            return new SQLiteRepository(databaseFilePath);
        }

        public override IDataRepository Open(string databaseFilePath, string password)
        {
            return new SQLiteRepository(databaseFilePath, password);
        }
    }
}
